src/solvers/flattening/boolbv.cpp

```diff
@@ -126,18 +126,28 @@ const bvt &boolbvt::convert_bv(const exprt &expr)
   // Iterators into hash_maps supposedly stay stable
   // even though we are inserting more elements recursively.
 
-  cache_result.first->second=convert_bitvector(expr);
+  const bvt &bv = convert_bitvector(expr);
+  CHECK_RETURN(!expected_width || bv.size() == *expected_width);
```

Comment:
I think you have to remove this, otherwise the invariant with diagnostics will never be triggered.